File: src/types/alert.ts
export interface AlertRule {
  id: string;
  server_id: string;
  name: string;
  metric_key: string;
  condition: 'gt' | 'lt' | 'eq';
  threshold: number;
  duration_sec: number;
  enabled: boolean;
  created_at: number;
}

export interface AlertEvent {
  id: string;
  rule_id: string;
  server_id: string;
  status: 'firing' | 'resolved';
  message: string;
  fired_at: number;
  resolved_at: number | null;
}

File: src-tauri/src/models/alert.rs
#![allow(dead_code)]

use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AlertRule {
    pub id: String,
    pub server_id: String,
    pub name: String,
    pub metric_key: String,
    pub condition: AlertCondition,
    pub threshold: f64,
    pub duration_sec: u32,
    pub enabled: bool,
    pub created_at: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum AlertCondition {
    Gt,
    Lt,
    Eq,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AlertEvent {
    pub id: String,
    pub rule_id: String,
    pub server_id: String,
    pub status: AlertStatus,
    pub message: String,
    pub fired_at: i64,
    pub resolved_at: Option<i64>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum AlertStatus {
    Firing,
    Resolved,
}
